### ISSUE TITLE
marvell-switch-tag differs from Linux DSA driver

### ISSUE THREAD

**Issue by luizluca** [the-tcpdump-group/tcpdump-htdocs#24]
Hello,

According to
https://github.com/the-tcpdump-group/tcpdump-htdocs/blob/3d93f29f1d7533da9b8b67675af2bfafa78db675/linktypes/marvell-switch-tag.html#L68-L90
https://github.com/the-tcpdump-group/tcpdump-htdocs/blob/3d93f29f1d7533da9b8b67675af2bfafa78db675/linktypes/marvell-switch-tag.html#L104-L113 published at https://www.tcpdump.org/linktypes/LINKTYPE_DSA_TAG_DSA.html , 

frame type code is formed by (b17<<2 | b16 <<1 | b12 ), something like:

```C
code = (dsa_header[1] & 0x3) | ((dsa_header[2] >> 4) & 1);
```

But the Linux driver code uses (b18<<2 | b17 <<1 | b12 ) or:

```C
code = (dsa_header[1] & 0x6) | ((dsa_header[2] >> 4) & 1);
```
See: https://github.com/torvalds/linux/blob/655fedaad36c0b31a6f6cb8f469b739b2359d74f/net/dsa/tag_dsa.c#L207

I could confirm that b17b16b12 does not match the IGMP code (0x02) the chip sent (the only one I got) while b18b17b12 does match. I cannot confirm that b18 and b12 are correct as I didn't have other command samples but IGMP but b17 is the second most significant bit.

**Comment by infrastation** [comment]
On a related note, the code point for Policy Mirror was wrong (it is correct now).

**Comment by infrastation** [comment]
In tcpdump's `print-dsa.c` the comment and the macro indeed define the code as `b18-b17-b12` (this seems to be one of the points discussed in pull request #13):
```c
/*

 * Mode: Forward, To_CPU, From_CPU, To_Sniffer
 * b29: (Source or Target) IEEE Tag Mode
 * b18: Forward's Src_Is_Trunk, To_CPU's Code[2], To_Sniffer's Rx_Sniff
 * b17: To_CPU's Code[1]
 * b16: Original frame's CFI
 * b12: To_CPU's Code[0]
 */

#define DSA_CODE(tag) ((TOK(tag, 1, 0x06, 1) << 1) | TOK(tag, 2, 0x10, 4))
```
Besides that, a couple fields have different names in the two packet diagrams. @vivien, would you like to proof-read the HTML page and make it consistent?

**Comment by infrastation** [comment]
Alright, the fix is trivial, it has been made in commit 6e68a19, thank you for reporting this issue!
